spec/models/click_api_spec.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe ClickApi do
+  context 'with required params' do
+    let(:click) do
+      ClickApi.new url: 'http://www.fda.gov/foo.html',
+                   query: 'my query',
+                   client_ip: '12.34.56.789',
+                   affiliate: 'nps.gov',
+                   position: '7',
+                   module_code: 'RECALL',
+                   vertical: 'web',
+                   user_agent: 'mozilla',
+                   access_key: 'basic_key'
+    end
+
+    describe '#valid?' do
+      it 'should be valid' do
+        expect(click.valid?).to be_truthy
+      end
+    end
+
+    describe '#log' do
+      it 'should log almost-JSON info about the click' do
```

Comment:
For this & the other specs, avoid using `should` in the description, i.e.:
`it 'logs almost-JSON info about the click' do`

http://www.betterspecs.org/#should